src/sas/qtgui/MainWindow/GuiManager.py

```diff
@@ -104,6 +105,18 @@ def __init__(self, parent=None):
         # Log the python version
         logging.info("Python: %s" % sys.version)
 
+        import subprocess, json
+        installed_modules = {'python': sys.version}
+
+        # Get python modules installed locally
+        installed_modules_json = json.loads(subprocess.check_output("pip list -l --format=json"))
```

Comment:
The following gets a string of pip installed packages in the current Python environment, without using a subprocess.

```
         try:
            from pip._internal.operations import freeze
        except ImportError:
            return ""

        return "\n".join(list(freeze.freeze()))
```
For conda installed packages you'd end up with something like:
`numpy @ file:///Users/runner/miniforge3/conda-bld/numpy_1629092076711/work`. To deal with those you'd have to use something like `subprocess.check_output("conda list")`, but conda may not be being used.

The pip approach works very well in a pyinstaller built standalone executable. When I build standalone executables of this type (perhaps sasview does the same), I always do it in a new virtual environment with all packages installed via a pip requirements file. Only using pip means the list is comprehensive. For the packages that need to be installed from source I do that with pip as well:
```
# in source dir
pip install .
# or
pip install git+https://.....
```

It seems that this short snippet would be more robust than the code in the `ModuleGatherer.py` file.

Reply:
Hi @andyfaff, thanks for the review

As far as I can tell, pip freeze / list etc get a list of modules installed on the system, either in a local virtual environment or globally. That’s is useful information to have and I aimed to make it accessible to the user. I see what you’re saying about the pip approach, and will also add the conda list function.

However I wanted to do something slightly different with the get_imported_modules function. My aim was to list every module that python imports, for example using an “import x” call, or a “from x import y as z” call; and get the exact version of that module. This is for 2 reasons. Firstly there is a chance that two versions of a module are present, one in the virtualenv and one on the system for example. I want to double check which one is being accessed. Secondly there may be modules in the virtual environment listed in pip that where not used in a session of SasView. Listing the imported modules will only show modules used in a session, so can help narrow down the search when attempting to bugfix. 

I couldn’t find any pip or built in function that does this accurately, hence the code in get_imported_modules.

I hope this made sense.

Reply:
That's a great point @Caddy-Jones. That could be a great help in tracking down failures that only occur for certain people I would guess?

Reply:
I hope so, I'm aiming to give the debuggers as much information as possible